Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/Indexer.kt

```diff
@@ -499,6 +499,26 @@ internal class NativeIndexImpl(val library: NativeLibrary, val verbose: Boolean
                 spelling = clang_getTypeSpelling(type).convertAndDispose().dropConstQualifier()
         )
 
+        CXType_Vector -> {
+            val elementCXType = clang_getElementType(type)
+            val elementType = convertType(elementCXType)
+            val size = clang_Type_getSizeOf(type)
+            val elemSize = clang_Type_getSizeOf(elementCXType)
+            val elementCount = size / elemSize
+            assert(elemSize * elementCount == size)
+
+            // Spelling example: `__attribute__((__vector_size__(4 * sizeof(float)))) const float`
+            // Re-generate spelling removing constness and typedefs to limit number of variants for bridge generator
+            // Supposed to be the same (i.e. natively compatible) as clang_getTypeSpelling(type) aka type.name
+            val spelling = "__attribute__((__vector_size__($size))) ${clang_getCanonicalType(elementCXType).name}"
```

Comment:
Does `clang_getCanonicalType` remove `const` modifier?
Is there a test that ensures it does?

Reply:
clang_getCanonicalType unwrap typedefs only, never remove qualifiers
Const (as well as volatile) in this context is a qualifier of the vector itself. Element type seems never having qualifiers. Even when the element type is declared with typedef the canonical type of element does not have qualifier. Makes sense, because elements can't have different qualifiers, so it is always propagated to the vector itself.
AFAIK this is not ruled by the C standard, and probably is implementation specific of clang parser.